multiversx_sdk_core/transaction_factories/token_operations_factory.py

```diff
@@ -451,342 +364,247 @@ def nft_create(
             arg_to_string(quantity_to_burn)
         ]
 
-        return self._create_transaction(
-            sender=manager,
-            receiver=manager,
-            nonce=transaction_nonce,
+        return self._create_transaction_intent(
+            sender=sender,
+            receiver=sender,
             value=None,
-            gas_price=gas_price,
-            gas_limit_hint=gas_limit,
             execution_gas_limit=self._config.gas_limit_esdt_nft_burn,
-            data_parts=parts,
-            guardian=guardian
+            data_parts=parts
         )
 
-    def _create_transaction(
+    def _create_transaction_intent(
         self,
         sender: IAddress,
         receiver: IAddress,
-        nonce: Optional[INonce],
-        value: Optional[ITransactionValue],
-        gas_price: Optional[IGasPrice],
-        gas_limit_hint: Optional[IGasLimit],
-        guardian: Optional[IAddress],
-        execution_gas_limit: IGasLimit,
+        value: Optional[int],
+        execution_gas_limit: int,
         data_parts: List[str]
-    ) -> Transaction:
+    ) -> TransactionIntent:
         payload = self._build_transaction_payload(data_parts)
-        version = TRANSACTION_VERSION_DEFAULT
-        options = TRANSACTION_OPTIONS_DEFAULT
 
-        has_guardian = True if guardian else False
-        gas_limit = gas_limit_hint or self._compute_gas_limit(payload, execution_gas_limit, has_guardian)
+        gas_limit = self._compute_gas_limit(payload, execution_gas_limit)
 
-        return Transaction(
-            chain_id=self._config.chain_id,
-            sender=sender,
-            receiver=receiver,
-            gas_limit=gas_limit,
-            gas_price=gas_price,
-            nonce=nonce or 0,
-            value=value or 0,
-            data=payload,
-            version=version,
-            options=options,
-            guardian=guardian
-        )
+        transaction_intent = TransactionIntent()
+
+        transaction_intent.sender = sender.bech32()
+        transaction_intent.receiver = receiver.bech32()
+        transaction_intent.gas_limit = gas_limit
+        transaction_intent.data = str(payload).encode()
+        transaction_intent.value = str(value) if value else "0"
+
+        return transaction_intent
 
     def _build_transaction_payload(self, parts: List[str]) -> TransactionPayload:
         data = ARGS_SEPARATOR.join(parts)
         return TransactionPayload.from_str(data)
 
-    def _compute_gas_limit(self, payload: TransactionPayload, execution_gas: IGasLimit, has_guardian: bool) -> IGasLimit:
+    def _compute_gas_limit(self, payload: TransactionPayload, execution_gas: int) -> int:
         data_movement_gas = self._config.min_gas_limit + self._config.gas_limit_per_byte * payload.length()
         gas = data_movement_gas + execution_gas
 
-        if has_guardian:
-            gas += self._config.extra_gas_limit_for_guarded_transactions
+        # add gas for guarded transactions by default
```

Comment:
As above, perhaps we can remove this (to be discussed & confirmed).

Reply:
It has been discussed, no extra gas is added anymore.